### ISSUE TITLE
False positive `no-member` for `enum.Enum` when imported with alias

### ISSUE THREAD

**Issue by bluefish6** [PyCQA/pylint#5776]
### Bug description

Enum is not being recognised as an Enum, when it's imported with an alias (to prevent confusion when also doing `from sqlalchemy.dialects.postgresql import ENUM`)

```
from enum import Enum as PyEnum


class MyEnum(PyEnum):
    ENUM_KEY = "enum_value"


foo = MyEnum.ENUM_KEY.value
```

### Command used

```shell
pylint example.py
```

### Pylint output

```shell
************* Module example
example.py:8:6: E1101: Instance of 'str' has no 'value' member (no-member)
```

### Expected behavior

No error

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.3
Python 3.8.10 (default, Nov 26 2021, 20:14:08) [GCC 9.3.0]
```

### OS / Environment

Ubuntu 20.04.3 LTS
